test/common/schemaform/FileField.unit.spec.jsx

```diff
@@ -0,0 +1,243 @@
+    );
+
+    expect(tree.everySubTree('label')[0].text()).to.contain('Upload');
+  });
+  it('should render files', () => {
+    const idSchema = {
+      $id: 'field'
+    };
+    const schema = {};
+    const uiSchema = fileUploadUI('Files');
+    const formData = [
+      {
+        confirmationCode: 'asdfds',
+        name: 'Test file name'
+      }
+    ];
+    const tree = SkinDeep.shallowRender(
+      <FileField
+          schema={schema}
+          uiSchema={uiSchema}
+          idSchema={idSchema}
+          formData={formData}
+          formContext={formContext}
+          onChange={f => f}
+          requiredSchema={requiredSchema}/>
+    );
+
+    expect(tree.subTree('li').text()).to.contain('Test file name');
+    expect(tree.subTree('.usa-button-outline').text()).to.contain('Edit');
+  });
+
+  it('should render uploading', () => {
+    const idSchema = {
+      $id: 'field'
+    };
+    const schema = {};
+    const uiSchema = fileUploadUI('Files');
+    const formData = [
+      {
+        uploading: true
+      }
+    ];
+    const tree = SkinDeep.shallowRender(
+      <FileField
+          schema={schema}
+          uiSchema={uiSchema}
+          idSchema={idSchema}
+          formData={formData}
+          formContext={formContext}
+          onChange={f => f}
+          requiredSchema={requiredSchema}/>
+    );
+
+    expect(tree.subTree('.va-growable-background').text()).to.contain('Uploading file...');
+    expect(tree.everySubTree('button')).to.be.empty;
+  });
+
+  it('should render error', () => {
+    const idSchema = {
+      $id: 'field'
+    };
+    const schema = {};
+    const uiSchema = fileUploadUI('Files');
+    const formData = [
+      {
+        errorMessage: 'asdfas'
+      }
+    ];
+    const errorSchema = {
+      0: {
+        __errors: [
+          'Bad error'
+        ]
+      }
+    };
+    const tree = SkinDeep.shallowRender(
+      <FileField
+          schema={schema}
+          uiSchema={uiSchema}
+          idSchema={idSchema}
+          errorSchema={errorSchema}
+          formData={formData}
+          formContext={formContext}
+          onChange={f => f}
+          requiredSchema={requiredSchema}/>
+    );
+
+    expect(tree.subTree('.va-growable-background').text()).to.contain('Bad error');
+  });
+
+  it('should not render upload button if over max items', () => {
+    const idSchema = {
+      $id: 'field'
+    };
+    const schema = {
+      maxItems: 1
+    };
+    const uiSchema = fileUploadUI('Files');
+    const formData = [
+      {
+        confirmationCode: 'asdfds',
+        name: 'Test file name'
+      }
+    ];
+    const tree = SkinDeep.shallowRender(
+      <FileField
+          schema={schema}
+          uiSchema={uiSchema}
+          idSchema={idSchema}
+          formData={formData}
+          formContext={formContext}
+          onChange={f => f}
+          requiredSchema={requiredSchema}/>
+    );
+
+    expect(tree.everySubTree('label')).to.be.empty;
```

Comment:
Is the label on the button? Just seems odd for the test to be `should not render upload button if over max items` and for us to not check for a button. :man_shrugging:

Apparently I'm all about finding buttons to push... 🤔